### PR TITLE
tso: remove local tso code part 2

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve?

Issue Number: Ref https://github.com/tikv/pd/issues/8802

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: errors.toml [hunk 1]
```diff
@@ -901,11 +901,6 @@ error = '''
 reset user timestamp failed, %s
 '''
 
-["PD:tso:ErrSetLocalTSOConfig"]
-error = '''
-set local tso config failed, %s
-'''
-
 ["PD:tso:ErrSyncMaxTS"]
```
Comment (rleungx) [inline errors.toml:904]: How about it?

Comment (okJiang) [reply errors.toml:904]: Let me remove it

File: pkg/tso/allocator_manager.go [hunk 2]
```diff
@@ -315,64 +314,6 @@ func (am *AllocatorManager) GetMember() ElectionMember {
...
-	// Check whether the dc-location number meets the upper limit 2**(LogicalBits-1)-1,
-	// which includes 1 global and 2**(LogicalBits-1) local
-	if len(clusterDCLocations) == int(math.Pow(2, MaxSuffixBits))-1 {
-		return errs.ErrSetLocalTSOConfig.FastGenByArgs("the number of dc-location meets the upper limit")
-	}
-	return nil
-}
-
 // GetClusterDCLocationsFromEtcd fetches dcLocation topology from etcd
 func (am *AllocatorManager) GetClusterDCLocationsFromEtcd() (clusterDCLocations map[string][]uint64, err error) {
```
Comment (rleungx) [inline pkg/tso/allocator_manager.go:318]: How about this one?

File: pkg/tso/allocator_manager.go [hunk 3]
```diff
@@ -413,26 +354,6 @@ func (am *AllocatorManager) GetDCLocationInfo(dcLocation string) (DCLocationInfo
...
-	log.Info("delete the dc-location key previously written in etcd",
-		logutil.CondUint32("keyspace-group-id", am.kgID, am.kgID > 0),
-		zap.Uint64("server-id", serverID))
-	go am.ClusterDCLocationChecker()
-	return nil
-}
-
 // GetClusterDCLocations returns all dc-locations of a cluster with a copy of map,
 // which satisfies dcLocation -> DCLocationInfo.
 func (am *AllocatorManager) GetClusterDCLocations() map[string]DCLocationInfo {
```
Comment (rleungx) [inline pkg/tso/allocator_manager.go:359]: ditto

File: pkg/tso/allocator_manager.go (hunk 2 above)
Comment (okJiang) [reply pkg/tso/allocator_manager.go:318]: This PR does not delete all the functions; it is already large enough. We just need to ensure that the deleted content does not affect normal operation. The remaining functions will be removed in subsequent PRs.

**Review by rleungx — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

**Pull request merged**